### DISCUSSION TITLE
[dashboard][ux]gracefully handle and display charts error with broken dataset on dashboard

### DISCUSSION THREAD

**Discussion by junlincc** [apache/superset#18504] [Ideas]
Charts' underling datasets could break in many ways, when they break, charts throw ugly errors both in explore and dashboard. The current state of the product still doesn't have a perfect way to handle different errors, lacking the explanation of cause explicitly, from the ui perspective, for business users. when users encounter error in Explore, it's visually less intrusive and scary, often time users have more patience to troubleshoot to understand what goes wrong; while multiple errors showing on a dashboard, users tend to be more frustrated and perceive it as that the product is broken. 

<img width="1771" alt="Screen Shot 2021-08-27 at 12 34 58 PM" src="https://user-images.githubusercontent.com/67837651/131171153-555528ae-9b5c-4612-a614-21e83cf9c441.png">

Proposed changes: 
for improving dashboard experience when underlying data source is problematic.
Instead of showing these scary error states which for some personas isn’t super friendly or actionable, could we show the last cached- data + charts but make it very clear that it’s in an error state.

For UI, we could display an error message at the bottom of last cached charts. the benefit of this change is to 
1) Preserve relatively useful content either as a reference for dashboard recreation using similar healthy dataset, or for minimum communication/presentation while underlying dataset is being fixed 
2) Less visually intrusive 
3) User can still access to the errors

i am not sure how difficult technically.

**Comment by etr2460** [comment]
> could we show the last cached- data + charts but make it very clear that it’s in an error state

I'm not sure how often this would be possible. If the chart was cached, then we'd show that already. If it's running the chart on the backend, then we wouldn't have cached data. If the chart itself is broken, then it might be confusing to a person who made a change to the chart but then still see the old ui.

Noting that these "scary error states" provide solutions behind the `see more` button, but we can't always expand it because the chart could be too small. Perhaps a good first step would be to expand the error by default when the chart is large enough?
